src/BridgeGame.js

```diff
@@ -1,20 +1,124 @@
+const { MAP, MOVING_SPACE } = require("./util/Constant");
+
 /**
  * 다리 건너기 게임을 관리하는 클래스
  */
 class BridgeGame {
+  #bridge;
+
+  #tryCount;
+
+  #movedSpace;
+
+  constructor(bridge) {
+    this.#bridge = bridge;
+    this.#tryCount = 1;
+    this.#movedSpace = [];
+  }
+
+  getBridge() {
+    return [...this.#bridge];
+  }
+
+  getTryCount() {
+    return this.#tryCount;
+  }
+
+  getMovedSpace() {
+    return [...this.#movedSpace];
+  }
+
+  /**
+   * 윗길, 아랫길로 각각 정해진 형식에 맞게 파싱한 값을 반환
+   * @returns {object}
+   */
+  getRoadStates() {
+    return {
+      upRoad: this.#getRoadState(MOVING_SPACE.UP),
+      downRoad: this.#getRoadState(MOVING_SPACE.DOWN),
+    };
+  }
+
+  /**
+   * 해당 길의 기록을 정해진 형식에 맞게 반환 (ex: "[ O |   | X ]")
+   * @param {string} road 윗길or아랫길 ("U" or "D")
+   * @returns {string}
+   */
+  #getRoadState(road) {
+    return (
+      MAP.START +
+      this.#parseMovedSpaceToRoadLog(road).join(MAP.SEPARATOR) +
+      MAP.END
+    );
+  }
+
+  /**
+   * 해당 길의 기록을 반환 (ex: ["O", " ", "X"])
+   * @param {string} road 윗길or아랫길 ("U" or "D")
+   * @returns {string[]}
+   */
+  #parseMovedSpaceToRoadLog(road) {
+    return this.#movedSpace.map((space, index) => {
+      if (space !== road) return MAP.EMPTY;
+      if (this.#isLastMoveFailed(index)) return MAP.FAIL;
+      return MAP.SUCCESS;
+    });
+  }
+
+  /**
+   * 마지막 이동이 성공인지 실패인지 알려주는 메서드
+   * @param {number} index 검사할 요소의 인덱스
+   * @returns {boolean}
+   */
+  #isLastMoveFailed(index) {
+    if (index !== this.#movedSpace.length - 1) return false;
+    return !this.isSuccessMoved();
+  }
+
+  /**
+   * 현재 이동한 칸이 성공인지 실패인지 알려주는 메서드
+   * @returns {boolean}
+   */
+  isSuccessMoved() {
+    const currentIndex = this.#movedSpace.length - 1;
+
+    if (this.#bridge[currentIndex] === this.#movedSpace[currentIndex])
+      return true;
+    return false;
```

Comment:
```suggestion
    return this.#bridge[currentIndex] === this.#movedSpace[currentIndex];
```
true와 false만 리턴하는 경우는 바로 조건식을 사용하여 리턴해주면 될 것 같습니다!

Reply:
네...ㅜ 앞으로 조건문 처리는 확실히 신경써서 해야겠습니다 ㅎㅎㅎ